### PR TITLE
[Merged by Bors] - Parse uint256 as decimal string

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

https://github.com/sigp/lighthouse/pull/2940 introduced a bug where we parsed the uint256 terminal total difficulty as a hex string instead of a decimal string. Fixes the bug and adds tests.

### CODE REVIEW

**Comment by michaelsproul** [comment]
Note on affected setups:

The bug only affects a Lighthouse v2.1.1 VC connected to a beacon node that _does not serve_ `TERMINAL_TOTAL_DIFFICULTY` on `/eth/v1/config/spec`. This is confirmed to include these beacon nodes:

- Lighthouse v2.0.1 or earlier

The following beacon nodes are compatible and _will not result in a panic_:

- Lighthouse v2.1.0
- Lighthouse v2.1.1
- Teku v22.1.0, or Infura
- Nimbus/v1.6.0-bef13b (from `testing.mainnet.beacon-api.nimbus.team`)

For anyone searching for this issue:

```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: FromHexError { inner: InvalidStringLength }', consensus/types/src/chain_spec.rs:694:10
```

File: consensus/types/src/chain_spec.rs [hunk 1]
```diff
@@ -689,9 +689,10 @@ fn default_bellatrix_fork_epoch() -> Option<MaybeQuoted<Epoch>> {
 }
 
 fn default_terminal_total_difficulty() -> Uint256 {
-    "115792089237316195423570985008687907853269984665640564039457584007913129638912"
-        .parse()
-        .unwrap()
+    Uint256::from_dec_str(
+        "115792089237316195423570985008687907853269984665640564039457584007913129638912",
+    )
+    .unwrap()
```
Comment (divagant-martian) [inline consensus/types/src/chain_spec.rs]: you could have this as a constant (`const fn default_terminal_total_difficulty() -> Uint256`) and in the test verify via this parsing that the constant is correct
```suggestion
    Uint256([
        18446744073709550592,
        18446744073709551615,
        18446744073709551615,
        18446744073709551615,
    ]);
```

File: consensus/types/src/chain_spec.rs [hunk 2]
```diff
@@ -1038,4 +1039,63 @@ mod yaml_tests {
             .expect("should have applied spec");
         assert_eq!(new_spec, ChainSpec::minimal());
     }
+
+    #[test]
+    fn test_defaults() {
```
Comment (divagant-martian) [inline consensus/types/src/chain_spec.rs:1049]: And somewhere here verify that
```rust
assert_eq!(
    Uint256::from_dec_str(
        "115792089237316195423570985008687907853269984665640564039457584007913129638912",
    ), 
    Ok(default_terminal_total_difficulty())
);
```

that way the value does not need to be parsed in runtime

**Review by divagant-martian — APPROVED** [review]
:relaxed:

File: consensus/types/src/chain_spec.rs [hunk 3]
```diff
@@ -1038,4 +1044,73 @@ mod yaml_tests {
...
+        SECONDS_PER_ETH1_BLOCK: 14
+        MIN_VALIDATOR_WITHDRAWABILITY_DELAY: 256
+        SHARD_COMMITTEE_PERIOD: 256
+        ETH1_FOLLOW_DISTANCE: 2048
+        INACTIVITY_SCORE_BIAS: 4
+        INACTIVITY_SCORE_RECOVERY_RATE: 16
+        EJECTION_BALANCE: 16000000000
+        MIN_PER_EPOCH_CHURN_LIMIT: 4
+        CHURN_LIMIT_QUOTIENT: 65536
+        # PROPOSER_SCORE_BOOST: 70
```
Comment (paulhauner) [inline consensus/types/src/chain_spec.rs]: We don't test this value later on, I'm not sure if that's intentional?

**Review by paulhauner — APPROVED** [review]
Looks good! I just wanted to check about `PROPOSER_SCORE_BOOST`.

File: consensus/types/src/chain_spec.rs (hunk 3 above)
Comment (pawanjay176) [reply consensus/types/src/chain_spec.rs]: Wanted to check just values that have a `serde(default)` tag. Will uncomment this.

**Review by paulhauner — APPROVED** [review]
LGTM! Nice one.

**Pull request closed without merging**